test/distributed/fsdp/test_fsdp_tp_integration.py

```diff
@@ -182,7 +187,14 @@ def _get_grads_as_flattened(
         returns the same value on all ranks.
         """
         local_grads_as_flattened = (
-            torch.cat([torch.flatten(param.grad) for param in model.parameters()])
+            torch.cat(
+                [
+                    torch.flatten(param.grad)
+                    if param.grad is not None
```

Comment:
Is this change needed for `use_orig_params=True` specifically?

Reply:
yes, for `use_orig_params=True` specifically